src/field/pokemon.ts

```diff
@@ -226,6 +229,19 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
     this.calculateStats();
   }
 
+
+  getNameToRender() {
+    try {
+      if (this.nickname) {
+        return decodeURIComponent(escape(atob(this.nickname)));
+      }
+      return this.name;
+    } catch (err) {
+      console.error(err);
```

Comment:
Please add some context to the error
```suggestion
      console.error(`Failed to decode nickname for ${this.name}`, err);
```